### PR TITLE
Use create3 in basic order creation

### LINKED ISSUE #1120: Use create3 instead of create in orders
Use [create3](https://github.com/0xsequence/create3) instead of create in orders creation (OrderFactory.sol)

### PR DESCRIPTION
**Pull request by Svetomech**
## Summary by CodeRabbit

- **New Features**
	- Enhanced order creation process with the integration of CREATE3.
	- New events for improved logging of order creation and salt usage.

- **Bug Fixes**
	- Improved error handling for order creation failures and salt reuse.

- **Documentation**
	- Added documentation comments for better clarity on function purposes and parameters.

- **Chores**
	- Updated package dependencies to include `@0xsequence/create3`.
	- Modified command signatures in the project management files to accommodate new parameters.

### CODE REVIEW

**Comment by mn13** [comment]
Compiler shows me warning
```
Warning (2018): Function state mutability can be restricted to pure
   --> test/BasicOrder.t.sol:427:5:
    |
427 |     function getTestSwapData() internal view returns (bytes memory) {
    |  
```

File: solidity/orders/script/CreateOrder.s.sol [hunk 1]
```diff
@@ -65,7 +65,8 @@ contract CreateOrder is Script {
         mockSlinkyPrecompile.setPrice(pricePair.base, pricePair.quote, thresholdPrice);
         vm.etch(IWARDEN_PRECOMPILE_ADDRESS, address(wPrecompile).code);
         vm.startBroadcast(broadcaster);
-        FACTORY.createOrder(orderData, maxKeychainFees, OrderType.Basic);
+        bytes32 salt = keccak256(abi.encodePacked(broadcaster, block.number));
+        FACTORY.createOrder(orderData, maxKeychainFees, OrderType.Basic, salt);
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply solidity/orders/script/CreateOrder.s.sol]: resolved by moving salt to arguments

File: solidity/orders/src/OrderFactory.sol [hunk 2]
```diff
@@ -44,24 +52,44 @@ contract OrderFactory is Ownable {
...
+        external
         returns (address order)
     {
+        if (usedSalts[salt]) {
+            revert SaltAlreadyUsed();
+        }
+
+        usedSalts[salt] = true;
+        emit SaltUsed(salt, msg.sender);
+
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply solidity/orders/src/OrderFactory.sol]: added ReentrancyGuard from OpenZeppelin

File: solidity/orders/package.json [hunk 3]
```diff
@@ -8,6 +8,7 @@
     "solhint": "^3.6.2"
   },
   "dependencies": {
+    "@0xsequence/create3": "https://github.com/0xsequence/create3",
```
[the remark this answers is not in the document]
Comment (Svetomech) [reply solidity/orders/package.json]: done

**Review by mn13 — APPROVED** [review]
LGTM!

**Pull request merged**